### PR TITLE
Extract descriptor fields for Instrument and use autovalue

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/AbstractInstrument.java [hunk 1]
```diff
@@ -92,21 +62,12 @@ public boolean equals(Object o) {
 
     AbstractInstrument that = (AbstractInstrument) o;
 
-    return name.equals(that.name)
-        && description.equals(that.description)
-        && unit.equals(that.unit)
-        && constantLabels.equals(that.constantLabels)
-        && labelKeys.equals(that.labelKeys);
+    return descriptor.equals(that.descriptor);
```
Comment (jkwatson) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/AbstractInstrument.java]: should this include the library info, so two libraries could have the same instrument descriptor, and not collide?

Comment (bogdandrutu) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/AbstractInstrument.java]: The instrument descriptor has to be unique inside a MeterSdk which is uniquely identified by the LibraryInfo, so I don't think we should include the LibraryInfo

Comment (jkwatson) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/AbstractInstrument.java]: ah, yes. that makes sense.  👍

**Comment by bogdandrutu** [comment]
@jkwatson PTAL :)

**Pull request merged**